### ISSUE TITLE
Media details: Delete button does not provide a description to VoiceOver

### ISSUE THREAD

**Issue by ctarda** [wordpress-mobile/WordPress-iOS#15002]
The Delete Button is not exposed to VoiceOver properly, it seems to be missing an accessibility label and/or description
<Img src="https://user-images.githubusercontent.com/2722505/94392260-636af280-018a-11eb-9684-522802c0da1b.PNG" width="350" alt="Screenshot of media details view with Delete button highlighted"/>

### Expected behavior
The Delete button in the media details view should provide a description to VoiceOver, so VoiceOver could read it as "button - delete"

### Actual behavior
At this moment, it just identifies itself as button, so VoiceOver reads it only was "button:

### Steps to reproduce the behavior
- Activate VoiceOver
- Navigate to My Site > Pick a site > Media > Select one image > Swipe left twice (to reach the delete icon in the navigation bar)
- Notice how the Delete icon is read only as "Button"

##### Tested on [device], iOS [version], WPiOS [version]
Tested on iPhone 6s+, iOS 13.7, WPiOS 15.7, es_ES
Tested on iPhone 11 Pro Max, iOS 14.0.1, WPiOS 15.7, en_CA
